File: Estudio de Teoría/DBManager/DBManager.cs
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Policy;
using System.Text;
using System.Threading.Tasks;

namespace EmpresaDBManager
{
    public class DBManager
    {
        private static DBManager instancia;
        private MySqlConnection conexion;

        private string servidor   = "server=base-de-datos-personal.cr94q8u6tnjv.us-east-1.rds.amazonaws.com;";
        private string usuario    = "user=admin;";
        private string baseDatos  = "database=Programacion_3;";
        private string puerto     = "port=3306;";
        private string contrasena = "password=base-de-datos-personal.,;";

        public MySqlConnection Conexion
        {
            get
            {
                string cadenaConexion = servidor + usuario + baseDatos + puerto + contrasena;
                conexion = new MySqlConnection(cadenaConexion);
                return conexion;
            }
        }

        private static void crearInstancia()
        {
            if (instancia == null)
                instancia = new DBManager();
        }

        public static DBManager Instancia
        {
            get
            {
                if (instancia == null)
                    crearInstancia();
                return instancia;
            }
        }
    }
}

File: Conexión Java y C#/Backend en Java/DBManager/src/empresa/config/DBManager.java
package empresa.config;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class DBManager {
    private static DBManager instance;
    private Connection connection;
    
    private final String url = "jdbc:mysql://" +
                               "base-de-datos-personal.cr94q8u6tnjv." +
                               "us-east-1.rds.amazonaws.com" +
                               ":3306/Programacion_3?useSSL=false";
    private final String user = "admin";
    private final String password = "base-de-datos-personal.,";
    
    private static void createInstance() {
        instance = new DBManager();
    }
    
    public static DBManager getInstance() {
        if (instance == null) createInstance();
        return instance;
    }

    public Connection getConnection() {
        try {
            Class.forName("com.mysql.cj.jdbc.Driver");
            connection = DriverManager.getConnection(url, user, password);
        } catch (ClassNotFoundException | SQLException ex){
            System.out.println(ex.getMessage());
        }
        
        return connection;
    }
}